docs/reference/errors.md

```diff
@@ -0,0 +1,566 @@
+---
+title: Error Messages
+description: Error messages identify problems with the syntax or meaning of a message.
+---
+
+You might encounter a number of different error messages while writing an
+MF2 message. Here's a guide to those errors, with an example for each one.
+
+> The presentation and wording of errors varies based on which implementation of MF2
+> you are using. This guide is based on the error descriptions in the
+> [MF2 specification](https://github.com/unicode-org/message-format-wg/blob/main/spec/errors.md);
+> for each one, the error name used by the JavaScript/TypeScript implementation
+> (which is used by the [playground](/playground)) is also provided.
+
+## Syntax errors
+
+JS name: `parse-error`
+
+Syntax errors occur when a message has errors that prevent the message processor
+from providing any more detailed information.
+
+<mf2-interactive>
+
+```mf2
+{|missing-brace|
+```
+
+</mf2-interactive>
+
+In this case, the error is reported as `parse-error` and provides a character
+offset within the message where the parser wasn't able to make progress.
+
+### Fixing the error
+
+Look over the message carefully with an eye towards errors like missing
+curly braces.
+
+Corrected message:
+
+<mf2-interactive>
+
+```mf2
+{|missing-brace|}
+```
+
+</mf2-interactive>
+
```

Comment:
The Rust parser that Nico and I wrote have much much much better error messages for syntax errors than any of the other implementations (because it's for editor integration). I am planning to integrate that into the playground. We can then suggest in this section to either use the editor extension (in which case you get good diagnostics + quick fixes right there), or to just use the playground to get a more helpful error if a different implementation gives a worse error.